docs/madr/decisions/073-spiffe-compliance.md

```diff
@@ -153,7 +157,711 @@ Future releases
+  certificateExpirationTime: "2025-06-03T17:35:33Z"
+  certificateRegenerations: 1
+  issuedBackend: ca-1 # name of MeshIdentity
+  lastCertificateRegeneration: "2025-06-02T17:35:33.624189876Z"
+  supportedBackends: # list of MeshTrust name
+  - ca-1
+...
+```
+2. User creates MeshTrust with CA of a new MeshIdentity `ca-2`
+3. Trust is propagated. We provide identity and keep the information(supported CA (Trust), and Identity) in DataplaneInsight
+```yaml
+...
+name: demo-client-65f94cb577-jxrrq-44wfv42db9z44xxf
+type: DataplaneInsight
+mTLS:
+  certificateExpirationTime: "2025-06-03T17:35:33Z"
+  certificateRegenerations: 1
+  issuedBackend: ca-1 # name of MeshIdentity
+  lastCertificateRegeneration: "2025-06-02T17:35:33.624189876Z"
+  supportedBackends: # list of MeshTrust name
+  - ca-1
+  - ca-2
+...
+```
+
+Based on this information we will update MeshService
+
+```yaml
+...
+name: second-test-server-b7dwz26bb9b4z676
+spec:
+  identities:
+  - type: SpiffeID
+    value: spiffe://example.com/zone/my-new-zone/ns/real-resource-ns/sa/default
+  ports:
+  - appProtocol: http
+    name: main
+    port: 80
+    targetPort: main
+  selector:
+    dataplaneTags:
+      app: second-test-server
+      k8s.kuma.io/namespace: real-resource-ns
+  state: Available
+status:
+  ...
+  tls:
+    status: Ready <- only if all DPPs have identity provided
+  vips:
+  - ip: 10.43.245.159
+type: MeshService
+```
+
+4. User create a new MeshIdentity `ca-2`
+5. We start deliver new identity provided by `ca-2`
+We reuse existing fields
+```yaml
+...
+name: demo-client-65f94cb577-jxrrq-44wfv42db9z44xxf
+type: DataplaneInsight
+mTLS:
+  certificateExpirationTime: "2025-06-03T17:35:33Z"
+  certificateRegenerations: 1
+  issuedBackend: ca-2 # name of MeshIdentity
+  lastCertificateRegeneration: "2025-06-02T17:35:33.624189876Z"
+  supportedBackends: # list of MeshTrust name
+  - ca-1
+  - ca-2
+...
+```
+Update SpiffeID based on identities
+```yaml
+...
+name: second-test-server-b7dwz26bb9b4z676
+spec:
+  identities:
+  - type: SpiffeID
+    value: spiffe://example.com/ns/real-resource-ns/sa/default
+  - type: SpiffeID
+    value: spiffe://example.fr/ns/real-resource-ns/sa/default
+  ports:
+  - appProtocol: http
+    name: main
+    port: 80
+    targetPort: main
+  selector:
+    dataplaneTags:
+      app: second-test-server
+      k8s.kuma.io/namespace: real-resource-ns
+  state: Available
+status:
+  ...
+  tls:
+    status: Ready
+  vips:
+  - ip: 10.43.245.159
+type: MeshService
+```
+
+6. User can remove MeshIdentity `ca-1`
+7. MeshTrust should be removed by the user
+
+![Rotation flow](assets/073/rotation-flow.png)
+
+#### Secret naming
+
+We want to use a [KRI](070-resource-identifier.md) for resource naming. Since `MeshIdentity` creates a separate resource (Secret), it should be named based on the KRI to simplify correlation between the `Secret` and the `MeshIdentity` configuration.
+
+Proposed name:
+
+```yaml
+kri_mi_mesh-1_us-east-2_kuma-system_identity-1
```

Comment:
not going to work on k8s as this is a secret so it needs to be the right DNS format